pykern/pkcli/ci.py

```diff
@@ -5,17 +5,47 @@
 :license: http://www.apache.org/licenses/LICENSE-2.0.html
 """
 from pykern.pkdebug import pkdp, pkdlog
+from pykern import pkunit
+from pykern.pkcli import test
+from pykern import pksetup
+import re
+
+
+_FILE_TYPE = re.compile(r'.py$')
+_EXCLUDE_FILES = re.compile( f"/{test.SUITE_D}/.*{pkunit.DATA_DIR_SUFFIX}/|/{test.SUITE_D}/.*_work/|/{pksetup.PACKAGE_DATA}/|pkdebug")
+_PRINT = re.compile('\s(pkdp)\(|\s(print)\(')
+
+
+def check_prints(exclude=_EXCLUDE_FILES):
+    """Recursively check repo for print and pkdp calls
+
+    Args:
+        exclude (regex): path pattern to exclude from check
+    """
+    from pykern import pkconst
+    from pykern import pkio
+
+    res = ""
+    for f in pkio.walk_tree(pkio.py_path(), _FILE_TYPE):
+        if re.search(exclude, str(f)):
+            continue
+        s = pkio.read_text(str(f))
+        for i, l in enumerate(s.split('\n')):
+            if re.search(_PRINT, l):
+                res += f'{f.basename} pkdp/print on line: {i + 1} :-> {l}\n'
+    if res:
+        pkconst.builtin_print(res)
+        raise AssertionError(f'\n{res}\nChecks fail due to pkdp/print present')
```

Comment:
Too many blank lines in the output. It's unnecessary to have a leading blank line. You don't know what comes before.  The "Checks fail due to ..." is implicit, so just output `res`. This command will be used in a context that will cause this command to be the last output. Thing "bigger" when designing error messages: where will it be seen and how will it be used? For a programmer running it, they'll see a failed output in a compile/test window. For CI, the test will cause the CI to stop. This is why "emojiful" output is useless in most cases. It looks pretty in one offs, but in the larger context it just creates noise for the developer to wade through. A good IDE will let you know a test/command fails.